In [12]:
import numpy as np

In [4]:
coefficients_uncensored = """0         2014.25      0.727468
1         2014.75      0.631149
2         2015.25      0.516993
3         2015.75      0.340047
4         2016.25      0.241202
5         2016.75      0.222907
6         2017.25      0.202406
7         2017.75      0.224765"""

In [15]:
np.around(np.mean([float(line.split()[-1]) for line in coefficients_uncensored.split("\n")]), 4)

0.3884

In [16]:
coefficients_censored = """0         2014.25      2.266497
1         2014.75      2.269250
2         2015.25      0.677349
3         2015.75      0.349733
4         2016.25      0.209036
5         2016.75      0.192601
6         2017.25      0.175276
7         2017.75      0.196672"""

In [17]:
np.around(np.mean([float(line.split()[-1]) for line in coefficients_censored.split("\n")]), 4)

0.7921

In [18]:
coefficients_multi_censored = """7            2011-10-01    -0.631077
8            2012-04-01    -0.868021
9            2012-10-01    -0.779859
10           2013-04-01    -0.350623
11           2013-10-01     0.185063
12           2014-04-01     0.333416
13           2014-10-01     0.551556"""

In [19]:
np.around(np.mean([float(line.split()[-1]) for line in coefficients_multi_censored.split("\n")]), 4)

-0.2228

In [20]:
correlations_uncensored = """0         2014.25              0.370674            -0.477786
1         2014.75              0.323026            -0.650315
2         2015.25              0.259765            -0.417765
3         2015.75              0.167928            -0.107088
4         2016.25              0.145978            -0.617214
5         2016.75              0.132136             0.608225
6         2017.25              0.111999             0.540958
7         2017.75              0.132898             0.294026"""

In [22]:
correlations_uncensored.split("\n")

['0         2014.25              0.370674            -0.477786',
 '1         2014.75              0.323026            -0.650315',
 '2         2015.25              0.259765            -0.417765',
 '3         2015.75              0.167928            -0.107088',
 '4         2016.25              0.145978            -0.617214',
 '5         2016.75              0.132136             0.608225',
 '6         2017.25              0.111999             0.540958',
 '7         2017.75              0.132898             0.294026']

In [23]:
[line.split() for line in correlations_uncensored.split("\n")]

[['0', '2014.25', '0.370674', '-0.477786'],
 ['1', '2014.75', '0.323026', '-0.650315'],
 ['2', '2015.25', '0.259765', '-0.417765'],
 ['3', '2015.75', '0.167928', '-0.107088'],
 ['4', '2016.25', '0.145978', '-0.617214'],
 ['5', '2016.75', '0.132136', '0.608225'],
 ['6', '2017.25', '0.111999', '0.540958'],
 ['7', '2017.75', '0.132898', '0.294026']]

In [24]:
np.around(np.mean([float(line.split()[-2]) for line in correlations_uncensored.split("\n")]), 4)

0.2056

In [21]:
np.around(np.mean([float(line.split()[-1]) for line in correlations_uncensored.split("\n")]), 4)

-0.1034

In [25]:
correlations_censored = """0         2014.25              0.534528            -0.468950
1         2014.75              0.511698            -0.321460
2         2015.25              0.281948            -0.468422
3         2015.75              0.140711            -0.101120
4         2016.25              0.106150            -0.592654
5         2016.75              0.090920             0.516670
6         2017.25              0.074183             0.633187
7         2017.75              0.093403             0.300720"""

In [27]:
np.around(np.mean([float(line.split()[-2]) for line in correlations_censored.split("\n")]), 4)

0.2292

In [28]:
np.around(np.mean([float(line.split()[-1]) for line in correlations_censored.split("\n")]), 4)

-0.0628

In [29]:
correlations_multi = """28     2011-10-01             -0.021554            -0.046119
29     2012-04-01              0.008036             0.010675
30     2012-10-01              0.038138             0.038619
31     2013-04-01              0.052188            -0.353919
32     2013-10-01             -0.055193             0.228119
33     2014-04-01             -0.051533            -0.058964
34     2014-10-01             -0.053551            -0.266037"""

In [30]:
np.around(np.mean([float(line.split()[-2]) for line in correlations_multi.split("\n")]), 4)

-0.0119

In [31]:
np.around(np.mean([float(line.split()[-1]) for line in correlations_multi.split("\n")]), 4)

-0.0639

| Model | Mean coefficient | Mean training r | Mean testing r |
| -- | -- | -- | -- |
| single tree, uncensored | 0.3884 | 0.2056 | -0.1034 |
| single tree, censored | 0.7921 | 0.2292 | -0.0628 |
| multi tree, censored | -0.2228 | -0.0119 | -0.0639 |